static/src/javascripts/projects/commercial/modules/messenger/passback.ts

```diff
@@ -0,0 +1,69 @@
+import { log } from '@guardian/libs';
+import { adSlotIdPrefix } from '../dfp/dfp-env-globals';
+import type { RegisterListener } from '../messenger';
+
+type PassbackMessagePayload = {
+	source: string;
+};
+
+/**
+ * A listener for 'passback' messages from ad slot iFrames
+ * Ad providers will invoke 'passback' to tell us they have not filled this slot
+ * In which case we need to refresh the slot with another ad
+ */
+const init = (register: RegisterListener): void => {
```

Comment:
Can we test this function? It looks like a few other files in `messenger` have corresponding `.spec` files.
Or does pushing the main business logic onto `window.googletag.cmd` complicate testing?

Reply:
You got me! :)

I will give this some further thought however I feel that the majority of the passback logic is so heavily involved with googletag that testing would mean creating very intricate mocks and devolve to asserting that we call googletag in a certain way.

I'm leaning away from such tests as I feel they become brittle and tied to implementation details. If we could get an e2e test around it though, that would be great!

I'll take a look at adding a test the logic around finding the slot from the iframe though 👍

Reply:
I did try adding a test but couldn't get a cross-window `postMessage` to work in jsdom.
I think we're keen to get this out asap so I will follow up in a separate PR!

Reply:
That makes sense that googletag makes testing this more trouble than it's worth. Thanks for looking into it!